### PR TITLE
Update mcr.cpp

### PR DESCRIPTION
**Pull request by HongzeLi98**

### CODE REVIEW

File: mcr.cpp [hunk 1]
```diff
@@ -1,98 +1,55 @@
 #include <iostream> 
 
-using namespace std; 
+using namespace std;
 
 bool isWin(char game[3][3]){ 
-
```
Comment (chesterzhang) [inline mcr.cpp:6]: format the code, in order to be easier understanded.

File: mcr.cpp [hunk 2]
```diff
@@ -1,98 +1,55 @@
...
+    char player2 = 'O'; 
+    bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first. 
+    cout << "X = Player 1" << endl << "O = Player 2" << endl; 
+    for (int n=0; n<9; n++){ 
+        turn = !turn;  // use the not-operator to change true to false or false to true. 
+        if (turn == false) 
+            cout << "Player 1: "; 
+        else 
+            cout << "Player 2: "; 
+        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
```
Comment (XiaRachel) [inline mcr.cpp:35]: Didn't check input correctness

**Review by chesterzhang — COMMENTED** [review]
format the code, in order to be easier understanded.

File: mcr.cpp [hunk 3]
```diff
@@ -1,98 +1,55 @@
...
+    if (game[0][0] == game[0][1] && game[0][1] == game[0][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
+    if (game[1][0] == game[1][1] && game[1][1] == game[1][2] && (game[1][0] == 'X' || game[1][0] == 'O')) win = true;
+    if (game[2][0] == game[2][1] && game[2][1] == game[2][2] && (game[2][0] == 'X' || game[2][0] == 'O')) win = true; 
+    // column 
+    if (game[0][0] == game[1][0] && game[1][0] == game[2][0] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
+    if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] == 'X' || game[0][1] == 'O')) win = true; 
+    if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true; 
+    // diagonal 
+    if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
+    if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true;
```
Comment (witten-wu) [inline mcr.cpp:17]: Is there a possibility of a tie? which means no win or lose.

**Review by XiaRachel — COMMENTED** [review]
Hi here

File: mcr.cpp [hunk 4]
```diff
@@ -1,98 +1,55 @@
...
+        if (turn == false) 
+            cout << "Player 1: "; 
+        else 
+            cout << "Player 2: "; 
+        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";  
+        cin >> i >> j; 
+        if (turn == false) 
+            game[i][j] = 'X'; 
+        else  
+            game[i][j] = 'O';
```
Comment (witten-wu) [inline mcr.cpp]: Can we consider adding a time limit to avoid either party not doing something?

Comment (HongzeLi98) [reply mcr.cpp]: I think it will lead to a new library, so we maybe consider introduce this feature in future versions.

File: mcr.cpp (hunk 3 above)
Comment (HongzeLi98) [reply mcr.cpp:17]: good idea

File: mcr.cpp [hunk 5]
```diff
@@ -1,98 +1,60 @@
...
+    cout << "X = Player 1" << endl << "O = Player 2" << endl; 
+    for (int n=0; n<9; n++){ 
+        turn = !turn;  // use the not-operator to change true to false or false to true. 
+        if (turn == false) 
+            cout << "Player 1: "; 
+        else 
+            cout << "Player 2: "; 
+        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";  
+        cin >> i >> j; 
+        if ((i==0||i==1||i==2) && (j==0||j==1||j==3))
```
Comment (XiaRachel) [inline mcr.cpp]: the range of i and j should be 1 to 3

File: mcr.cpp [hunk 6]
```diff
@@ -1,98 +1,55 @@
...
-if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
-
-if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true; 
-
-return win; 
-
-} 
-
+    bool win = false; 
+    if (game[0][0] == game[0][1] && game[0][1] == game[0][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true;
```
Comment (chesterzhang) [inline mcr.cpp:7]: Can you simplify the if statements? You can see some of them is unnecessary.

**Review by chesterzhang — COMMENTED** [review]
Add some comments in win function

File: mcr.cpp (hunk 6 above)
Comment (HongzeLi98) [reply mcr.cpp:7]: No problem, sir.

File: mcr.cpp (hunk 1 above)
Comment (HongzeLi98) [reply mcr.cpp:6]: Sir, yes sir.

**Pull request merged**